official-templates/base/Dockerfile

```diff
@@ -134,26 +131,34 @@ RUN apt-get update --yes && \
 
 # Install pip
 RUN curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py && \
-    python3.7 get-pip.py && \
     python3.8 get-pip.py && \
     python3.9 get-pip.py && \
     python3.10 get-pip.py && \
-    python3.11 get-pip.py
+    python3.11 get-pip.py && \
+    rm get-pip.py
 
 # Get the latest pip for all python versions
-RUN python3.7 -m pip install --upgrade pip && \
-    python3.8 -m pip install --upgrade pip && \
+RUN python3.8 -m pip install --upgrade pip && \
     python3.9 -m pip install --upgrade pip && \
     python3.10 -m pip install --upgrade pip && \
     python3.11 -m pip install --upgrade pip
 
 # Install virtualenv
-RUN python3.7 -m pip install virtualenv && \
-    python3.8 -m pip install virtualenv && \
+RUN python3.8 -m pip install virtualenv && \
     python3.9 -m pip install virtualenv && \
     python3.10 -m pip install virtualenv && \
     python3.11 -m pip install virtualenv
 
+# Install Jupyter and related packages (always with Python 3.10)
```

Comment:
Not sure if this should be part of the base layer, but open to thoughts.

Reply:
from a devex perspective, I support adding Jupyter